File: scripts/scala/scala_2.10_da_prog_funcional_as_novas_features_scala/src/com/github/diegopacheco/sandbox/scala/partial/PartialMain.scala
package com.github.diegopacheco.sandbox.scala.partial

object PartialMain extends App {
	
  def soma(a:Int,b:Int):Int = { 
    a + b 
  } 
  
  val soma2 = soma(2, _:Int)
  
  println( soma2(2) ) 
  
}

File: scripts/scala/scala_2.10_da_prog_funcional_as_novas_features_java/src/com/github/diegopacheco/sandbox/java/partial/PartialMain.java
package com.github.diegopacheco.sandbox.java.partial;

public class PartialMain {
	
	public static Integer soma(Integer a, Integer b){
		return a + b;
	}
	
	interface Partial{
		public Integer get();
	}
	
	public static Integer soma2(Partial b){
		return soma(2,b.get());
	}
	
	public static void main(String[] args) {
		System.out.println(
			
			soma2(new Partial(){
				public Integer get() {
					return 2;
				}
			})
			
		);
	}
	
}

